src/Algolia.Search.Test/RetryStrategyTest/RetryStrategyTest.cs

```diff
@@ -128,14 +129,14 @@ public void TestRetryStrategyMultiThread(CallType callType)
             Task task1 = Task.Run(() =>
             {
                 var hosts = retryStrategy.GetTryableHost(callType);
-                retryStrategy.Decide(hosts.ElementAt(0), 200, false);
+                retryStrategy.Decide(hosts.ElementAt(0), new AlgoliaHttpResponse { HttpStatusCode = 200});
                 Console.WriteLine(Thread.CurrentThread.Name);
             });
 
             Task task2 = Task.Run(() =>
             {
                 var hosts = retryStrategy.GetTryableHost(callType);
-                retryStrategy.Decide(hosts.ElementAt(0), 500, false);
+                retryStrategy.Decide(hosts.ElementAt(0), new AlgoliaHttpResponse { HttpStatusCode = 500});
```

Comment:
Maybe we could add a `task3` to assert that network errors set the host down:
```
            Task task3 = Task.Run(() =>
            {
                var hosts = retryStrategy.GetTryableHost(callType);
                retryStrategy.Decide(hosts.ElementAt(0), new AlgoliaHttpResponse { IsNetworkError = true });
            });
```

And then assert that the host.up will be == 2.

Reply:
👍